Report the same info from general nginx requests

elenatanasoiu opened the pull request:
Log the same amount of info for general nginx requests as for the nginx reverse proxy.

elenatanasoiu committed:
Report the same info for general nginx requests

For the reverse proxy requests we log a bigger list of info. We
should do the same for general nginx requests.

Since we use log_json in default.conf, removing the additional log
config here will make it default to using log_json.

surminus commented:
I wonder, given that we already define `log_format` elsewhere, if we could reuse that? Rather than defining it twice, we could set the `access_log` parameter in the same way we set it for the site?[1]

I'm not entirely sure how ordering works here in terms of loading configuration, but it's relatively straight forward to test using Docker. 

[1] https://github.com/futurelearn/dockerfiles/blob/2690e3f0e7d28462f40c2c2a3f126a239eae8f45/nginx-proxy/default.conf#L17

surminus reviewed (approved):
Looks good.

I pushed a commit which may help with testing[1] these kinds of changes in the future.

[1] https://github.com/futurelearn/dockerfiles/commit/e82ef2a588a663a2b091a3c4c56f38fe5b2e927f

surminus commented on nginx-proxy/nginx.conf:
```diff
@@ -15,9 +15,29 @@ http {
-    log_format  main  '$remote_addr - $remote_user [$time_local] "$request" '
-                      '$status $body_bytes_sent "$http_referer" '
-                      '"$http_user_agent" "$http_x_forwarded_for"';
+    log_format main escape=json '{'
```
Does it work if `log_json` is referenced after that particular config is loaded [here](https://github.com/futurelearn/dockerfiles/pull/9/files#diff-b1dc99521b923cc30faedd827b5fbcbdL47)? 

This may avoid defining it twice.

elenatanasoiu committed:
Add missing env var

In order to test using docker-compose, we need API_SERVER_NAME to
be passed as an env var.

surminus reviewed (approved):
Assuming this boots OK, go for it!

elenatanasoiu commented:
Tested and it responds to requests correctly.
